wikimedia/wikipedia-ios
# [HOLD] Native Source Editor - Bold/italic syntax highlighting support

**Phabricator:** 
https://phabricator.wikimedia.org/T347925 (bold/italic/bolditalic syntax highlighting, syntax highlighting toggle support)

### Notes
_Note: This PR is dependent on #4672. Marking as hold while we wait on that._

This PR adds the bold / italics syntax highlighting to the text editor. It also adds support to the "show syntax highlighting" toggle in the popover.

### Test Steps
1. Launch app on Staging iOS 17 (TextKit 2). Go to a large article. Tap "Edit" in the navigation bar to edit the full article wikitext.
2. Confirm you can see bold and italic syntax highlighting as you scroll.
3. Add bold (''') or italic ('') wikitext around words. Confirm they update properly.
4. Turn syntax highlighting off from toggle. Confirm syntax highlighting disappears.
5. Repeat steps 1 - 4 on iOS 15 or 16 (TextKit 1).
